### PR TITLE
Python 3.7 support

### PR DESCRIPTION
**Pull request by dices**
Renamed pymodbus.server.async to pymodbus.server.asynchronous
Modified asyncio.async to use asyncio.ensure_future
Added missing prompt_toolkit dependency to requirements-docs.txt
Updated references in documentation
Updated examples
Same as previous pr but started from dev to remove the commit I did not want to have in the pull request

### CODE REVIEW

**Comment by dices** [comment]
The 3.7 environment is broken, cannot download python 3.7 zip

**Comment by dhoomakethu** [comment]
@dices looks like travis still does not really support 3.7 (refer https://github.com/travis-ci/travis-ci/issues/9815). I will keep this PR open till we get an official support and builds passing. There is also another PR with python3.7 support (@cbergmiller ). Will take some time to go through the code and review.

**Comment by dhoomakethu** [comment]
@dices  on a second thought, these changes introduce a lot of backward incompatibility with respect to async clients. Also the naming convention seems off with the sync client )`pymodbus.client.sync` vs `pymodbus.client.asynchronous`. I ran some bigquery to check the percentage of 3.7 downloads and I find the number very less to get this to master.
How about creating a new dev branch for 3.7 and work with that till we come to a common conclusion ?

screen shot 2018-09-27 at 10 33 28

**Comment by dices** [comment]
Sounds ok with me. I was also a bit worried about the backward
compatibility. Once you have the dev branch ready I can make the push
request against that one. Maybe think of a shorter name instead of the long
asynchronous one. I just wanted to contribute since I am using 3.7 and
pymodbus was not working on it.

**Comment by dhoomakethu** [comment]
@dices Thanks, I have created a new branch `dev3.7` , please rebase this PR against it.

File: pymodbus/client/asynchronous/__init__.py [hunk 1]
```diff
@@ -31,4 +31,18 @@
...
+installed = is_installed('twisted')
+if installed:
+    # Import deprecated async client only if twisted is installed #338
+    from pymodbus.client.async.deprecated.async import *
+else:
+    import logging
+    logger = logging.getLogger(__name__)
+    logger.warning("Not Importing deprecated clients. "
+                   "Dependency Twisted is not Installed")
+>>>>>>> 7980f087a0ecbf24ca049849799b365892082029:pymodbus/client/async/__init__.py
```
Comment (tracernz) [inline pymodbus/client/asynchronous/__init__.py]: Just a heads up, you managed to commit the merge conflict here. :wink:

Comment (dices) [reply pymodbus/client/asynchronous/__init__.py]: Shame on me for doing the merge so late during the night. I did it in  the wrong branch too :)

This one will be closed the new one on the correct branch has been sent

**Pull request closed without merging**